### PR TITLE
BIP 43: Reserve purpose codes 10001-19999 for SLIPs

### PR DESCRIPTION
**Pull request by luke-jr**
@slush0 @prusnak 

cc @Arachnid

### CODE REVIEW

**Comment by Arachnid** [comment]
:+1:

**Comment by prusnak** [comment]
I don't mind

**Comment by luke-jr** [comment]
@prusnak is that an ACK, or should I wait for @slush0 ?

**Comment by jprichardson** [comment]
NACK on deterministic scheme for alts. BIPs are for Bitcoin only and should not dictate a scheme for alts: https://github.com/bitcoin/bips/pull/76#issuecomment-48480728 Plus, the wallets have determined a scheme using BIP44.

ACK on purpose code allocation for SLIPs though.

**Comment by luke-jr** [comment]
It doesn't dictate anything, merely reserves a path for altcoin *standards processes* to use. BIP 44 does not suffice for this purpose. For example, see https://github.com/ethereum/EIPs/pull/600

**Comment by jprichardson** [comment]
> merely reserves a path for altcoin standards processes to use. 

They can already do this. They can use any purpose constant they want. 

> BIP 44 does not suffice for this purpose. For example, see ethereum/EIPs#600

It suffices just fine - 10 wallets use it: https://github.com/ethereum/EIPs/issues/84#issuecomment-292324521 There was discussion about this over a year ago: https://github.com/ethereum/EIPs/issues/84

BIP44/43 are sufficient as-is. The only thing this PR should be concerned about is the reservation of the purpose constants for SLIPs.

**Comment by Arachnid** [comment]
> They can already do this. They can use any purpose constant they want.

Without coordination, however, the chance of more than one coin using the same constant is very high.

> It suffices just fine - 10 wallets use it

The link you provide illustrates exactly why it _doesn't_ work well: the BIP44 scheme is poorly suited to coins that don't use UTXO mechanisms, resulting in different wallets interpreting it differently and using incompatible derivation paths. The 10 wallets you mention are using 3 different mutually incompatible derivation paths.

File: bip-0043.mediawiki [hunk 1]
```diff
@@ -53,6 +55,17 @@ and calling themselves BIPxx compatible, we suggest that software which needs
...
+
+Purpose code 43 is reserved for usage by non-Bitcoin cryptocoins.
+In this case, the level immediately below the purpose code is hardened derivation using the index for the cryptocoin specified by [[https://github.com/satoshilabs/slips/blob/master/slip-0044.md|SLIP-0044]].
+The details of levels below this coin type are to be entirely decided by the standard process of the specific cryptocoin, but it is recommended that the very next level be a sub-purpose code with a similar use as this BIP's purpose code.
+
+So that the recommended use for non-Bitcoin cryptocoins is:
+
+<pre>
+m / 43' / coin_type' / subpurpose' / ...
+</pre>
```
Comment (webmaster128) [inline bip-0043.mediawiki:68]: Could this PR get merged if someone pulls out this specification to some other place, such that you do no more than reserving 43 here?

**Comment by webmaster128** [comment]
If we write a SLIP-0043 with the suggested content, we can reduce this PR to a single sentence:

> Purpose codes 13/43/48 are taken by SLIP-0013/SLIP-0043/SLIP-0048 and the range 10001 to 19999 is reserved for other [[https://github.com/satoshilabs/slips|SLIPs]].

This would delegate what the purpose 43 does and at the same time point to the usage of 13 and 48.

**Comment by prusnak** [comment]
ACK for `Purpose codes from 10001 to 19999 are reserved for [[https://github.com/satoshilabs/slips|SLIPs]].`

nACK for the rest. Altcoins are not and will not follow this.

**Pull request merged**